app/uk/gov/hmrc/voapropertylinking/connectors/modernised/ExternalPropertyLinkApi.scala

```diff
@@ -56,6 +56,14 @@ class ExternalPropertyLinkApi @Inject()(
         ).flatten
     )
 
+  def getMyOrganisationsPropertyLinksCount()(implicit request: RequestWithPrincipal[_]): Future[Option[Int]] = {
+   (http.GET[Option[PropertyLinksWithAgents]](myOrganisationsPropertyLinksUrl))
```

Comment:
your adding logic into the connector layer, can we please not do this. we have an established pattern that the services should do the logic. also we know we are calling getMyOrganisationsPropertyLinks.

Reply:
also why is property links with agents optional return? we have no custom logic for not found do we?

Reply:
I was thinking we will be replace connector with new End point which returns only Count Int
In this case, we will only make changes to connector point to New End point and no need to touch other part of the code

Reply:
I am thinking new End point is going to be ready soon and no need to add logic to service layer and make changes to both Connector and Service layer if new end point is ready

Reply:
About Option, the getMyOrganisationsPropertyLink is using Option, I just followed it :)